### ISSUE TITLE
Codegen output not implements codec::Decode

### ISSUE THREAD

**Issue by fzzr-** [paritytech/subxt#315]
Some `codegen` output not implements `codec::Decode`.

To reproduce there's two runtimes metadata:
[pontem.scale.zip](https://github.com/paritytech/subxt/files/7506851/pontem.scale.zip)

```rust
#[subxt::subxt(runtime_metadata_path = "nodes/pontem.scale")]
// or similar for pontem-xcmp.scale
pub mod pontem {}
```

Error: the trait `Decode` is not implemented for `BTreeMap<pontem::runtime_types::polkadot_parachain::primitives::Id, MessageQueueChain>`

<details> 
	<summary type="button">build log errors</summary>
	<pre><code>...
 error[E0277]: the trait bound `BTreeMap<pontem::runtime_types::polkadot_parachain::primitives::Id, MessageQueueChain>: Decode` is not satisfied
	--> src/main.rs:31:1
	 |
 31 | #[subxt::subxt(runtime_metadata_path = "nodes/pontem-xcmp.scale")]
	 | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	 | |
	 | the trait `Decode` is not implemented for `BTreeMap<pontem::runtime_types::polkadot_parachain::primitives::Id, MessageQueueChain>`
	 | in this procedural macro expansion
	 |
	::: ~/.cargo/git/checkouts/subxt-d8146022e5587204/78fd590/macro/src/lib.rs:39:1
	 |
 39 | pub fn subxt(args: TokenStream, input: TokenStream) -> TokenStream {
	 | ------------------------------------------------------------------ in this expansion of `#[subxt::subxt]`
	 |
	 = help: the following implementations were found:
				  <BTreeMap<K, V> as Decode>
 note: required by a bound in `subxt::StorageEntry::Value`
	--> ~/.cargo/git/checkouts/subxt-d8146022e5587204/78fd590/src/storage.rs:50:17
	 |
 50 |     type Value: Decode;
	 |                 ^^^^^^ required by this bound in `subxt::StorageEntry::Value`
 
 error[E0277]: the trait bound `BTreeMap<pontem::runtime_types::polkadot_parachain::primitives::Id, Vec<pontem::runtime_types::polkadot_core_primitives::InboundHrmpMessage<u32>>>: Decode` is not satisfied
	 --> src/main.rs:31:1
	  |
 31  | #[subxt::subxt(runtime_metadata_path = "nodes/pontem-xcmp.scale")]
	  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
	  | |
	  | the trait `Decode` is not implemented for `BTreeMap<pontem::runtime_types::polkadot_parachain::primitives::Id, Vec<pontem::runtime_types::polkadot_core_primitives::InboundHrmpMessage<u32>>>`
	  | in this procedural macro expansion
	  |
	 ::: ~/.cargo/git/checkouts/subxt-d8146022e5587204/78fd590/macro/src/lib.rs:39:1
	  |
 39  | pub fn subxt(args: TokenStream, input: TokenStream) -> TokenStream {
	  | ------------------------------------------------------------------ in this expansion of `#[subxt::subxt]`
	  |
	  = help: the following implementations were found:
					<BTreeMap<K, V> as Decode>
 note: required by a bound in `parity_scale_codec::Decode::decode`
	 --> ~/.cargo/registry/src/github.com-1ecc6299db9ec823/parity-scale-codec-2.3.1/src/codec.rs:284:15
	  |
 284 |     fn decode<I: Input>(input: &mut I) -> Result<Self, Error>;
	  |                  ^^^^^ required by this bound in `parity_scale_codec::Decode::decode`
	</code></pre>
</details>

**Comment by kvinwang** [comment]
workaround:
```rust
#[subxt::subxt(runtime_metadata_path = "nodes/pontem.scale")]
pub mod pontem {}
const _: () = {
    use pontem::runtime_types::polkadot_parachain::primitives::Id;

    impl PartialEq for Id {
        fn eq(&self, other: &Self) -> bool {
            self.0 == other.0
        }
    }

    impl Eq for Id {

    }

    impl PartialOrd for Id {
        fn partial_cmp(&self, other: &Self) -> Option<std::cmp::Ordering> {
            self.0.partial_cmp(&other.0)
        }
    }

    impl Ord for Id {
        fn cmp(&self, other: &Self) -> std::cmp::Ordering {
            self.0.cmp(&other.0)
        }
    }
};
```

**Comment by ascjones** [comment]
Alternatively you might be able to use `substitute_type` and define a local `Id` type e.g.

```
#[derive(Eq, PartialEq, Encode, Decode)]
pub struct ParachainId ( .. );

#[subxt::subxt(runtime_metadata_path = "nodes/pontem.scale")]
pub mod pontem {
   #[subxt(substitute_type = "polkadot_parachain::primitives::Id")]
    use super::ParachainId;
}
```

That might work.

**Comment by jsdw** [comment]
I believe that this will be closed by #456; you'll no longer need the `Ord` trait when that lands.
